### PR TITLE
WIP: Compile complex assignments to RIR bytecode

### PR DESCRIPTION
**Pull request by vogr**
WIP branch, please do not merge!

### CODE REVIEW

**Comment by o-** [comment]
Here are some stats for your PR:

* WARNING: the longest CI job test_release_2 took 3.05h
* The real_thing suite regressed by 0.94
* Overall benchmarks regressed by 1.0

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1284909302&job_ids[]=1282826854

File: rir/src/ir/Compiler.cpp [hunk 1]
```diff
@@ -647,118 +661,315 @@ bool compileSpecialCall(CompilerContext& ctx, SEXP ast, SEXP fun, SEXP args_,
...
+            if (Compiler::profile)
+                cs << BC::recordCall();
+
+            // prepare x, yk, z as promises
+            LoadArgsResult load_arg_res;
+            SEXP farrow_args = CDR(farrow_ast);
+
+            // Load the value of x as an eager promise
+
+            compileLoadOneArg(ctx, farrow_args, ArgType::EAGER_PROMISE,
```
Comment (o-) [inline rir/src/ir/Compiler.cpp:900]: the answer is probably "yes", but does `x` really need to be promise wrapped?

**Review by o- — APPROVED** [review]
Thanks a lot! lgtm

File: rir/src/ir/Compiler.cpp (hunk 1 above)
Comment (vogr) [reply rir/src/ir/Compiler.cpp:900]: This is not something I had tried before. I thought all arguments needed to be passed in promises: but if I try to pass the `x` and `z` (value) arguments as raw values, the (short) tests still pass

**Comment by o-** [comment]
Here are some stats for your PR:

* WARNING: the longest CI job test_release_1 took 3.54h
* The real_thing suite regressed by 0.93
* Overall benchmarks improved by 1.01

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1290942287&job_ids[]=1290886347

**Comment by o-** [comment]
Here are some stats for your PR:

* WARNING: the longest CI job tests_debug2 took 2.9h
* The real_thing suite regressed by 0.96
* The simple suite improved by 1.02
* Overall benchmarks improved by 1.01

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1294663936&job_ids[]=1290886347

**Pull request merged**